pkg/agent/openflow/pipeline_windows.go

```diff
@@ -140,3 +140,113 @@ func (c *client) l3FwdFlowToRemoteViaRouting(localGatewayMAC net.HardwareAddr, r
+		serviceConnectionTrackTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+			Action().CT(false, serviceConnectionTrackTable.GetNext(), SNATCtZone).NAT().CTDone().
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Done(),
+		// This flow sends Service request packet from antrea-gw0 with ServiceCTMark to serviceConntrackCommitTable.
+		connectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			MatchCTMark(ServiceCTMark).
+			MatchRegMark(FromGatewayRegMark).
+			Action().GotoTable(serviceConntrackCommitTable).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Done(),
+		// This flow sends Service packet to serviceConntrackCommitTable.
+		connectionTrackCommitTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+			MatchCTMark(ServiceCTMark).MatchCTStateNew(false).MatchCTStateTrk(true).
+			Action().GotoTable(hairpinSNATTable).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			Done(),
+		// This flow SNATs +new request packet.
+		serviceConnectionTrackCommitTable.BuildFlow(priorityHigh).MatchProtocol(binding.ProtocolIP).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			MatchInPort(config.HostGatewayOFPort).
+			MatchCTStateNew(true).
+			MatchCTStateTrk(true).
+			MatchCTStateDNAT(true).
+			MatchRegMark(SNATVirtualIPRegMark).
+			Action().LoadRegMark(HairpinRegMark).
+			Action().CT(true, serviceConnectionTrackCommitTable.GetNext(), SNATCtZone).
+			SNAT(&binding.IPRange{StartIP: config.ServiceGWHairpinIPv4, EndIP: config.ServiceGWHairpinIPv4}, nil).
+			CTDone().
+			Done(),
+		// This flow SNATs -new request packet.
+		serviceConnectionTrackCommitTable.BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			MatchInPort(config.HostGatewayOFPort).
+			MatchCTStateNew(false).
+			MatchCTStateTrk(true).
+			Action().CT(false, serviceConnectionTrackCommitTable.GetNext(), SNATCtZone).NAT().
+			CTDone().
+			Done(),
+	}
+}
+
+func (c *client) l3FwdFlowClusterIPServices(localGatewayMAC net.HardwareAddr, category cookie.Category) []binding.Flow {
+	return []binding.Flow{
+		// This flow sends DNATed request packet (from host or local Pods) to snatTable.
+		c.pipeline[l3ForwardingTable].BuildFlow(priorityLow).MatchProtocol(binding.ProtocolIP).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			MatchCTStateTrk(true).
+			MatchCTMark(ServiceCTMark).
+			MatchRegMark(RewriteMACRegMark).
+			Action().GotoTable(snatTable).
+			Done(),
+		// This flow modifies destination MAC of request packet from local Pods and sends it to l2ForwardingCalcTable.
+		c.pipeline[snatTable].BuildFlow(priorityNormal).MatchProtocol(binding.ProtocolIP).
+			Cookie(c.cookieAllocator.Request(category).Raw()).
+			MatchRegMark(FromLocalRegMark).
+			Action().SetDstMAC(localGatewayMAC).
+			Action().GotoTable(l2ForwardingCalcTable).
+			Done(),
```

Comment:
ditto, the first two flows are trying to do same thing as `l3FwdServiceDefaultFlowsViaGW`. Please check whether other flows are duplicate or not.
@hongliangl could you review this as well to avoid repeated flows and efforts?

Reply:
I didn't find that the 2 flows equal to the one in `l3FwdServiceDefaultFlowsViaGW`. Maybe it needs consideration with other flows.